miniml/compiler/compile.scm

```diff
@@ -1754,15 +1827,8 @@
       ))
    (('MTypedef tdefs)
     (fold (lambda (tdef env) (compile-type env (car tdef) (cdr tdef))) env tdefs))
-   (('MStruct name l)
-    (let* ((mark (lambda (e) (vhash-map (lambda (k v) (cons #f (cdr v))) e)))
-           (modenv (mkenv (mark (env-get-vars env))
-                          (mark (env-get-constrs env))
-                          (mark (env-get-fields env))
-                          (mark (env-get-modules env))))
-           (nenv (compile-defs modenv l)))
-      (env-replace-module env name (cons #t nenv))
-      ))
+   (('MModule name args mod)
+    (env-replace-module env name (cons #t (compile-functor env args mod))))
```

Comment:
Given the way it is used, `compile-functor` should be called `compile-module` I think.